GCC 10 requires two non-backward-compatible options

Available GCC versions on C4 are: 6, 7, 8 (default), 9, 10
Available GCC versions on C5: 10, 11 (default)

Using the current `ncrc-gcc.mk` template with GCC 11 on T5, @mcallic2 encountered errors such as:

```

Error: Type mismatch between actual argument at (1) and actual argument at (2) (REAL(8)/LOGICAL(4)).
/lustre/f2/dev/Caitlyn.Mcallister/am5/2022.01/am5p0p1_2022.04-alpha3/src/FMS/mpp/include/mpp_comm_mpi.inc:355:44:

  355 |       if( mpp_npes().GT.1 ) call MPI_BCAST( data, length*size(data(:)), MPI_CHARACTER, from_rank, peset(n)%id, error )
      |                                            1
Error: Type mismatch between actual argument at (1) and actual argument at (2) (CHARACTER(*)/LOGICAL(4)).
make[1]: *** [Makefile:151: mpp.o] Error 1
make[1]: Leaving directory '/lustre/f2/dev/gfdl/Caitlyn.Mcallister/am5/2022.01/am5p0p1_2022.04-alpha3/ncrc5.gcc-debug/exec/fms'
make: *** [Makefile:14: fms/libfms.a] Error 2
```

According to @thomas-robinson (and @rem1776 ), FFLAGS needs to have `-fallow-invalid-boz -fallow-argument-mismatch` for GCC versions 10 and later.

They are are described in the GCC 10 release notes (https://gcc.gnu.org/gcc-10/changes.html):

* Mismatches between actual and dummy argument lists in a single file are now rejected with an error. Use the new option -fallow-argument-mismatch to turn these errors into warnings; this option is implied with -std=legacy. -Wargument-mismatch has been removed.
* The handling of a BOZ literal constant has been reworked to provide better conformance to the Fortran 2008 and 2018 standards. In these Fortran standards, a BOZ literal constant is a typeless and kindless entity. As a part of the rework, documented and undocumented extensions to the Fortran standard now emit errors during compilation. Some of these extensions are permitted with the -fallow-invalid-boz option, which degrades the error to a warning and the code is compiled as with older gfortran.

These two options are not backward compatible, so we'll need to have two GCC templates or drop support for GCC versions before 10.

--- a/templates/ncrc-gcc.mk
+++ b/templates/ncrc-gcc.mk
@@ -81,7 +81,7 @@ FPPFLAGS := $(INCLUDES)
 FPPFLAGS += $(shell nf-config --fflags)
 
 # Base set of Fortran compiler flags
-FFLAGS := -fcray-pointer -fdefault-real-8 -fdefault-double-8 -Waliasing -ffree-line-length-none -fno-range-check
+FFLAGS := -fcray-pointer -fdefault-real-8 -fdefault-double-8 -Waliasing -ffree-line-length-none -fno-range-check -fallow-argument-mismatch
 
 # Flags based on perforance target (production (OPT), reproduction (REPRO), or debug (DEBUG)
 FFLAGS_OPT = -O2 -fno-expensive-optimizations

